cmd/geth/snapshot.go

```diff
@@ -173,8 +174,14 @@ func pruneState(ctx *cli.Context) error {
 		Cachedir:  stack.ResolvePath(config.Eth.TrieCleanCacheJournal),
 		BloomSize: ctx.Uint64(utils.BloomFilterSizeFlag.Name),
 	}
-	pruner, err := pruner.NewPruner(chaindb, prunerconfig,
-		pruner.WithTriesInMemory(ctx.Uint64(utils.TriesInMemoryFlag.Name)))
+	pruner, err := pruner.NewPruner(chaindb, prunerconfig, pruner.CombinedOptions{
+		PrunerOptions: []pruner.PrunerOption{
+			pruner.WithTriesInMemory(ctx.Uint64(utils.TriesInMemoryFlag.Name)),
+		},
+		SnapshotOptions: []snapshot.SnapshotOption{
+			snapshot.SetCapLimit(int(ctx.Uint64(utils.TriesInMemoryFlag.Name))),
```

Comment:
Please help confirm that it is intended to reuse `utils.TriesInMemoryFlag` as `snapshot.capLimit`?

Reply:
Brilliant point!
All occurrences of 'capLimit' in the current code are set to the same value as 'TriesInMemory'. However, based on the current mechanism, this doesn't seem necessary. Therefore, we need to decide whether to introduce an additional separate configuration option specifically for 'capLimit' after confirming some details.